hqmedia.upload_controller.js

```diff
@@ -286,6 +286,10 @@ function BaseHQMediaUploadController (uploader_name, marker, options) {
         if (!self.allowCloseDuringUpload) {
             self.allowClose = false;
         }
+        if (!self.isMultiFileUpload) {
+            var newExtension = '.' + self.filesInQueueUI[0].get('name').split('.').pop().toLowerCase();
```

Comment:
What if there is no extension? I believe this will add a "." to the end of the file in that case. Is that what we want? Maybe we don't care?